### PR TITLE
Add doc on how to uninstall ECK

### LINKED ISSUE #1434: Document operator uninstall
We should document how to cleanly uninstall the operator from a cluster.

### PR DESCRIPTION
**Pull request by thbkrkr**
Resolves #1434.

### CODE REVIEW

**Comment by anyasabo** [comment]
Deleting the CRDs deletes the CRs by definition (<--- turbo puns), so they should be able to just delete the all in one file. Also I think we need to add a section on deleting the webhook configuration since that is not contained within the file (and has no owner ref).

**Comment by thbkrkr** [comment]
> they should be able to just delete the all in one file

Oh! You're right, but that doesn't seem to work every time.

Sometimes it's completely stuck and the CR and underlying resources are not deleted:

```bash
> kubectl delete -f https://download.elastic.co/downloads/eck/0.9.0/all-in-one.yaml
customresourcedefinition.apiextensions.k8s.io "apmservers.apm.k8s.elastic.co" deleted
customresourcedefinition.apiextensions.k8s.io "elasticsearches.elasticsearch.k8s.elastic.co" deleted
customresourcedefinition.apiextensions.k8s.io "trustrelationships.elasticsearch.k8s.elastic.co" deleted
customresourcedefinition.apiextensions.k8s.io "kibanas.kibana.k8s.elastic.co" deleted
clusterrole.rbac.authorization.k8s.io "elastic-operator" deleted
clusterrolebinding.rbac.authorization.k8s.io "elastic-operator" deleted
namespace "elastic-system" deleted
statefulset.apps "elastic-operator" deleted
secret "webhook-server-secret" deleted
serviceaccount "elastic-operator" deleted
```

Looks like the statefulset is deleted too quickly. 

If I delete only the CRD, the CR are well deleted.
If I delete first the operator statefulset and after the CRD, the CRD deletion is stuck and CR are not removed.

So, unfortunately, I think it's not reliable enough, at least for the moment.

Is there something wrong with what we are doing?

> I think we need to add a section on deleting the webhook configuration

Yes!

**Comment by anyasabo** [comment]
>If I delete only the CRD, the CR are well deleted.
>If I delete first the operator statefulset and after the CRD, the CRD deletion is stuck and CR are not removed.

Finalizers strike again! So your method should work. The alternative would be deleting the CRDs, then deleting the operator resources (rbac, secrets, sset, webhook, etc). But they are more or less the same.

File: docs/uninstall.asciidoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+This should delete all underlying Elasticsearch, Kibana or APM Server resources (pods, secrets, services, etc.).
+
+Then you can uninstall the operator itself:
+
+[source,shell]
+----
+kubectl delete -f https://download.elastic.co/downloads/eck/0.9.0/all-in-one.yaml
+----
+
+And terminate with the webhook configuration:
```
Comment (anyasabo) [inline docs/uninstall.asciidoc]:
```suggestion
And remove the webhook configuration:
```

File: docs/uninstall.asciidoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+== Uninstall ECK
+
+Before you uninstall the operator, you must to uninstall the Elastic resources that it has created:
+
+[source,shell]
+----
+kubectl delete elastic --all --all-namesapces
+----
+
+This should delete all underlying Elasticsearch, Kibana or APM Server resources (pods, secrets, services, etc.).
```
Comment (anyasabo) [inline docs/uninstall.asciidoc]:
```suggestion
This deletes all underlying Elasticsearch, Kibana, and APM Server resources (pods, secrets, services, etc.).
```

File: docs/uninstall.asciidoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
+[id="{p}-uninstall"]
+== Uninstall ECK
+
+Before you uninstall the operator, you must to uninstall the Elastic resources that it has created:
```
Comment (sebgl) [inline docs/uninstall.asciidoc]:
```suggestion
Before uninstalling the operator, uninstall Elastic resources it has created:
```

Comment (anyasabo) [inline docs/uninstall.asciidoc]:
```suggestion
Before you uninstall the operator, you must remove all Elastic resources:
```

**Comment by thbkrkr** [comment]
> Will you be backporting this to 0.9 as well?

Good point. Yes, I think it's worth it.

File: docs/uninstall.asciidoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Before uninstalling the operator, remove all Elastic resources:
+
+[source,shell]
+----
+kubectl delete elastic --all --all-namesapces
+----
+
+This deletes all underlying Elasticsearch, Kibana, and APM Server resources (pods, secrets, services, etc.).
+
+Then you can uninstall the operator itself:
```
Comment (alaudazzi) [inline docs/uninstall.asciidoc]: Then, you can uninstall the operator:

**Pull request merged**